int32[] realIndex
contextRepresentation[] listInput
